src/main/scala/io/sdkman/changelogs/JavaMigrations.scala

```diff
@@ -385,25 +385,6 @@ class JavaMigrations {
     Seq(Linux64, Windows, MacOSX).foreach(platform => removeVersion(candidate = "java", version = "12.0.1-zulu", platform))
   }
 
-  @ChangeSet(order = "163", id = "163-add_graalvm_19_1_1", author = "wololock")
```

Comment:
I'm not 100% sure but (@marc0der will confirm it) I think you shouldn't remove previous migrations. At least it's the first time I see removing an old migration when adding a new new.

Reply:
@ilopmar I've asked for the trimming down of obsolete changesets as the `JavaMigrations` file is getting very bloated.

I was also toying with the idea of deprecating `JavaMigrations` and branching out into every JDK distribution having it's own migration file. Then we could refrain from deleting changesets.

In this case, `19.1.1` is not obsolete though as it will still remain in production for a while. @eddumelendez can we keep for now?

Reply:
Oracle now has a [version roadmap](https://www.graalvm.org/docs/release-notes/version-roadmap/) for graalvm. It's clear that what should be a _minor_ version in semantic versioning, it's in fact a _major_ version for graalvm (they use a weird [calver](http://calver.org/) `YY.MAJOR.MICRO`).

So a proposal could be, keep the last 3 major versions plus the last LTS patch.